worker/backup_processor.go

```diff
@@ -310,3 +323,49 @@ func writeKVList(list *bpb.KVList, w io.Writer) error {
 	_, err = w.Write(buf)
 	return err
 }
+
+func checkAndGetDropOp(key []byte, l *posting.List, readTs uint64) (*pb.DropOperation, error) {
+	isDropOpKey, err := x.IsDropOpKey(key)
+	if err != nil || !isDropOpKey {
+		return nil, err
+	}
+
+	vals, err := l.AllValues(readTs)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot read value of dgraph.drop.op")
+	}
+	switch len(vals) {
+	case 0:
+		// do nothing, it means this one was deleted with S * * deletion.
+		// So, no need to consider it.
+		return nil, nil
+	case 1:
+		val, ok := vals[0].Value.([]byte)
+		if !ok {
+			return nil, errors.Errorf("cannot convert value of dgraph.drop.op to byte array")
+		}
+		// A dgraph.drop.op record can have values in only one of the following formats:
+		// * DROP_ALL;
+		// * DROP_DATA;
+		// * DROP_ATTR;attrName
+		// So, accordingly construct the *pb.DropOperation.
+		dropOp := &pb.DropOperation{}
+		dropInfo := strings.Split(string(val), ";")
+		if len(dropInfo) != 2 {
+			return nil, errors.Errorf("Unexpected value: %s for dgraph.drop.op", val)
+		}
+		switch dropInfo[0] {
+		case "DROP_ALL":
+			dropOp.DropOp = pb.DropOperation_ALL
+		case "DROP_DATA":
+			dropOp.DropOp = pb.DropOperation_DATA
+		case "DROP_ATTR":
+			dropOp.DropOp = pb.DropOperation_ATTR
+			dropOp.DropValue = dropInfo[1]
+		}
+		return dropOp, nil
+	default:
+		// getting more than one values for a non-list predicate is an error
+		return nil, errors.Errorf("found multiple values for dgraph.drop.op")
```

Comment:
You might want to add the vals here so that it helps in debugging.

Reply:
Done